File: rd-net/Test.Cross/Static/CrossTestAllEntities.cs
using demo;
using JetBrains.Collections.Viewable;
using JetBrains.Diagnostics;
using JetBrains.Rd.Base;

namespace Test.RdCross
{
    // ReSharper disable once UnusedMember.Global
    public static class CrossTest_AllEntities
    {
      internal static void FireAll(DemoModel model, ExtModel extModel)
        {
            model.Boolean_property.Set(false);

            model.Boolean_array.Set(new[] {false, true, false});

            var scalar = new MyScalar(
                false,
                50,
                32000,
                1000000000,
                -2000000000000000000,
                3.14f,
                -123456789.012345678,
                byte.MaxValue - 1,
                ushort.MaxValue - 1,
                uint.MaxValue - 1,
                ulong.MaxValue - 1,
                MyEnum.net,
                Flags.anyFlag | Flags.netFlag,
                MyInitializedEnum.hundred
            );

            // ReSharper disable once UnusedVariable
            // ReSharper disable once InconsistentNaming
            var (_bool, _byte, _short, _int, _long, _float, _double, _unsigned_byte, unsigned_short, _unsigned_int, _unsigned_long, _my_enum, _flags, initializedEnum) = scalar;
            var (first, second) = new ComplicatedPair(new Derived("First"), new Derived("Second"));

            model.Scalar.Set(scalar);

            model.Set.Add(50);

            model.MapLongToString.Add(50, "C#");

            const string valA = "C#";
            const string valB = "protocol";

            model.Interned_string.Set(valA);
            model.Interned_string.Set(valA);
            model.Interned_string.Set(valB);
            model.Interned_string.Set(valB);
            model.Interned_string.Set(valA);

            var derived = new Derived("C# instance");
            model.Polymorphic.Set(derived);

            var openDerived = new OpenDerived("C# instance open derived string", "C# instance open string");
            model.Polymorphic_open.Set(openDerived);

            var openClass = new OpenClass("c# test");
            model.OpenClassProperty.Set(openClass);
            openClass.String.Set("property");

            model.Enum.Value = MyEnum.net;
            extModel.Checker.Fire();
        }

      internal static void CheckConstants()
        {
            Assertion.Assert(DemoModel.const_toplevel, "DemoModel.const_toplevel");
            Assertion.Assert(ConstUtil.const_enum == MyEnum.@default, "ConstUtil.const_enum == MyEnum.@default");
            Assertion.Assert(ConstUtil.const_string == "const_string_value",
                "ConstUtil.const_string == 'const_string_value'");
            Assertion.Assert(demo.Base.const_base == 'B', "Base.const_base == 'B'");
        }
    }
}

File: rd-kt/rd-cross/src/main/kotlin/com/jetbrains/rd/cross/statics/CrossTest_AllEntities.kt
package com.jetbrains.rd.cross.statics

import com.jetbrains.rd.util.reactive.fire
import demo.*
import java.util.*

@Suppress("ClassName")
@ExperimentalUnsignedTypes
object CrossTest_AllEntities {
    internal fun checkConstants() {
        assert(DemoModel.const_toplevel) { "DemoModel.const_toplevel" }
        assert(ConstUtil.const_enum == MyEnum.default) { "ConstUtil.const_enum == MyEnum.default" }
        assert(ConstUtil.const_string == "const_string_value") { "ConstUtil.const_string == \"const_string_value\"" }
        assert(Base.const_base == 'B') { "ConstUtil.const_string == \"const_string_value\"" }
    }

    internal fun fireAll(model: DemoModel, extModel: ExtModel) {
        model.boolean_property.set(false)

        model.boolean_array.set(booleanArrayOf(true, false, false))

        val scalar = MyScalar(false,
            13,
            32000,
            1_000_000_000,
            -2_000_000_000_000_000_000,
            3.14f,
            -123456789.012345678,
            UByte.MAX_VALUE.minus(1u).toUByte(),
            UShort.MAX_VALUE.minus(1u).toUShort(),
            UInt.MAX_VALUE.minus(1u),
            ULong.MAX_VALUE.minus(1u),
            MyEnum.kt,
            EnumSet.of(Flags.anyFlag, Flags.ktFlag),
            MyInitializedEnum.hundred
        )

        model.scalar.set(scalar)

        model.ubyte.set(UByte.MAX_VALUE.minus(13u).toUByte())

        model.ubyte_array.set(ubyteArrayOf(13u.toUByte(), (-1).toUByte()))

        model.list.add(1)
        model.list.add(3)

        model.set.add(13)

        model.mapLongToString[13] = "Kotlin"

        val valA = "Kotlin"
        val valB = "protocol"

        model.interned_string.set(valA)
        model.interned_string.set(valA)
        model.interned_string.set(valB)
        model.interned_string.set(valB)
        model.interned_string.set(valA)

        val derived = Derived("Kotlin instance")
        model.polymorphic.set(derived)

        val openDerived = OpenDerived("Kotlin instance open derived string ", "Kotlin instance open string")
        model.polymorphic_open.set(openDerived)

        val openClass = OpenClass("field")
        openClass.string.set("Kotlin test")

        model.date.set(Date(13_000)) // Thu Jan 01 03:00:13 MSK 1970

        model.enum.set(MyEnum.kt)

        extModel.checker.fire()

        model.struct_with_open_field.set(StructWithOpenStructField(OpenStructInField("", "", 123, "", "")))
    }
}